jdaviz/configs/imviz/plugins/catalogs/catalogs.py

```diff
@@ -324,23 +328,58 @@ def plot_selected_points(self):
                                                            getattr(y, 'value', y))
 
     def vue_zoom_in(self, *args, **kwargs):
-        """This function will zoom into the image based on the selected points"""
+
+        self.zoom_to_selected(return_bounding_box=True)
+
+    def zoom_to_selected(self, padding=0.02, return_bounding_box=False):
+        """
+        Zoom to a region containing the currently selected points in the catalog.
+
+        Parameters
+        ----------
+        padding : float, optional
+            A fractional value representing the padding around the bounding box
+            of the selected points. It is applied as a proportion of the largest
+            dimension of the viewer's current limits. Defaults to 0.02.
+        return_bounding_box : bool, optional
+            If True, returns the bounding box of the zoomed region as
+            ((x_min, x_max), (y_min, y_max)). Defaults to False.
+
+        Returns
+        -------
+        tuple of tuple, optional
+            If there are activley selected rows, and `return_bounding_box` is
+            True, returns a tuple containing the bounding
+            box coordinates: ((x_min, x_max), (y_min, y_max)).
+            Otherwise, returns None.
+
+        """
+
+        viewer = self.app._jdaviz_helper._default_viewer
+
         selected_rows = self.table.selected_rows
 
+        if not len(selected_rows):
+            return
+
         x = [float(coord['x_coord']) for coord in selected_rows]
         y = [float(coord['y_coord']) for coord in selected_rows]
 
+        limits = viewer.state._get_reset_limits()
+        max_dim = max((limits[1] - limits[0]), (limits[3] - limits[2]))
+        padding = max_dim * padding
+
         # this works with single selected points
         # zooming when the range is too large is not performing correctly
-        x_min = min(x) - 50
-        x_max = max(x) + 50
-        y_min = min(y) - 50
-        y_max = max(y) + 50
+        x_min = min(x) - padding
+        x_max = max(x) + padding
+        y_min = min(y) - padding
+        y_max = max(y) + padding
 
-        self.app._jdaviz_helper._default_viewer.set_limits(
-            x_min=x_min, x_max=x_max, y_min=y_min, y_max=y_max)
+        viewer.set_limits(x_min=x_min, x_max=x_max, y_min=y_min, y_max=y_max)
 
-        return (x_min, x_max), (y_min, y_max)
+        if return_bounding_box:
```

Comment:
Why the extra check here? What is wrong with always returning bounding box?

Reply:
it felt weird to have it return the bounding box in the notebook cell if you're just directing it to zoom, personally

Reply:
Using a trailing semicolon would suppress the output in notebook but I see your point. Thanks for the clarifications!